enhancements/update/cvo-techpreview-manifests.md

```diff
@@ -0,0 +1,139 @@
+---
+title: CVO TechPreview Manifests
+authors:
+  - "@deads2k"
+reviewers:
+  - "@wking"
+approvers:
+  - "@sdodson"
+  - "@wking"
+api-approvers: # in case of new or modified APIs or API extensions (CRDs, aggregated apiservers, webhooks, finalizers)
+  - @sttts
+creation-date: 2021-11-15
+last-updated: 2021-11-15
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - slack thread somewhere
+see-also:
+replaces:
+superseded-by:
+---
+
+# CVO TechPreview Manifests
+
+## Summary
+
+[TechPreviewNoUpgrade](https://github.com/openshift/api/blob/be1be0e89115702f8b508d351c4f5c9a16e5ae95/config/v1/types_feature.go#L32-L34)
+is the canonical way to enable tech preview features.
+It is the setting that all operators use to enable Tech Preview features.
+The CVO will honor this setting on a per-manifest basis, so that tech preview manifests are never present on
+non-tech preview clusters.  
+
+## Motivation
+
+When introducing tech preview operators, the current state of the art (before this enhancement) is to have the 
+operator build an inert mode that detects if tech preview is not enable and then does not take action.
+This approach results in having additional resources, including the clusteroperator/tech-preview itself, 
+installed on every cluster, including non-tech-preview upgrades. 
+This also means that currently, if the tech preview does not progress to stable, the component has to keep "dead" manifests
+around and marked for deletion by the CVO on upgraded clusters.
+By having the CVO honor tech preview per manifest, we can avoid fanning out inert logic to every tech preview operator,
+we can avoid exposing tech preview operators to customers via clusteroperators, and we can avoid creating unnecessary
+resources like namespaces, roles, rolebindings, serviceacounts, and deployments.
+
+### Goals
+
+1. CVO honors an "only create or update this manifest in clusters with tech preview enabled".
+
+### Non-Goals
+
+1. Bootstrap rendering for tech preview operators.
+2. 4.y upgrades.  TechPreviewNoUpgrade explicitly disallows upgrades from 4.y to 4.y+1.
+3. Cleanup of manifests created as TechPreviewNoUpgrade.  TechPreviewNoUpgrade explicitly disallows being unset and this
+    is enforced using validation on the apiserver.
+4. Allow manifests to be removed or un-reconciled if TechPreviewNoUpgrade is *not* set.  There are a few case (mostly
+    around CRDs), where this capability is useful, but it is not as common as needing to create something additional.
+
+## Proposal
+
+Add an annotation that can be set on CVO manifests called `config.openshift.io/feature-gate` with one legal/honored value
```

Comment:
This functionality is similar to [profile's `include.release.openshift.io/[identifier]=true`][1].  Can we use `config.release.openshift.io/featue-gate` or some such to stay in that same `release.openshift.io` namespace for manifest annotations aimed at CVO consumption?

[1]: https://github.com/openshift/enhancements/blob/ce4d303db807622687159eb9d3248285a003fabb/enhancements/update/cluster-profiles.md#cluster-profile-annotation

Reply:
> This functionality is similar to [profile's `include.release.openshift.io/[identifier]=true`](https://github.com/openshift/enhancements/blob/ce4d303db807622687159eb9d3248285a003fabb/enhancements/update/cluster-profiles.md#cluster-profile-annotation). Can we use `release.openshift.io/featue-gate` or some such to stay in that same `release.openshift.io` namespace for manifest annotations aimed at CVO consumption?

Sure, updating.